packages/ckeditor5-dev-changelog/src/generatechangelog.ts

```diff
@@ -25,20 +25,40 @@ import { getRepositoryUrl } from './utils/external/getrepositoryurl.js';
 import { getNewChangelog } from './utils/getnewchangelog.js';
 import { removeChangesetFiles } from './utils/removechangesetfiles.js';
 import { removeScope } from './utils/removescope.js';
+import { InternalError } from './errors.js';
 
 export async function generateChangelog(
-  config: RepositoryConfig & GenerateChangelog & { noWrite?: false | undefined }
+	config: RepositoryConfig & GenerateChangelog & { noWrite?: false | undefined }
 ): Promise<void>;
 
 export async function generateChangelog(
-  config: RepositoryConfig & GenerateChangelog & { noWrite: true }
+	config: RepositoryConfig & GenerateChangelog & { noWrite: true }
 ): Promise<string>;
 
+/**
+ * Wrapper function that provides error handling for the changelog generation process.
+ */
+export async function generateChangelog(
+	options: RepositoryConfig & GenerateChangelog
+): Promise<string | void> { // eslint-disable-line @typescript-eslint/no-invalid-void-type
+	try {
+		return await main( options );
```

Comment:
```suggestion
		return main( options );
```

Reply:
Sorry! The proposal changes how the `generateChangelog()` handles an error returned by the `main()` function. We must use `await`.